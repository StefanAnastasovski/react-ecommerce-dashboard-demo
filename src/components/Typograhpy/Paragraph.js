import React from "react";

import { Typography } from "@mui/material";

const Paragraph = ({ children, ...otherProps }) => {
  return (
    <Typography component="p" {...otherProps}>
      {children}
    </Typography>
  );
};

export default Paragraph;
