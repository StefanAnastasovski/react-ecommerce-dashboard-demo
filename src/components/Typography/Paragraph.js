import React from "react";

import { Typography } from "@mui/material";

const Paragraph = ({ children, ...otherProps }) => {
  return (
    <Typography paragraph mb={0} {...otherProps}>
      {children}
    </Typography>
  );
};

export default Paragraph;
