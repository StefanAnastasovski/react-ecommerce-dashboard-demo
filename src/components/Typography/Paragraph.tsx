import React from "react";

import { Typography } from "@mui/material";
import ParagraphType from "../../@types/ParagraphType";

const Paragraph = ({ children, styleProps, otherProps }: ParagraphType) => {
  return (
    <Typography paragraph mb={0} sx={{ ...styleProps }} {...otherProps}>
      {children}
    </Typography>
  );
};

export default Paragraph;
