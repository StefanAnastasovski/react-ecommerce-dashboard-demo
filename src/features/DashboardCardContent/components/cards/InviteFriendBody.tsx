import React from "react";

import { Box } from "@mui/material";

import {
  LINK_N_TEXT_FONT_SIZE,
  PRIMARY_GREEN_COLOR_ACTIVE,
  FONT_WEIGHT_700,
} from "../../../../data/constants";

import Paragraph from "../../../../components/Typography/Paragraph";

const InviteFriendBody = () => {
  const styledCompleted = {
    fontSize: LINK_N_TEXT_FONT_SIZE,
  };

  const highlightedText = {
    color: PRIMARY_GREEN_COLOR_ACTIVE,
    fontSize: LINK_N_TEXT_FONT_SIZE,
    fontWeight: FONT_WEIGHT_700,
  };

  return (
    <>
      <Paragraph styleProps={{...styledCompleted, padding: "1em 0"}}>
        <Box component="span" {...highlightedText} pr={0.5}>
          Receive 50 products
        </Box>
        by inviting a friend who subscribes to a plan. Your friend will receive
        a 30% discount coupon valid for any plan.
      </Paragraph>
    </>
  );
};

export default InviteFriendBody;
