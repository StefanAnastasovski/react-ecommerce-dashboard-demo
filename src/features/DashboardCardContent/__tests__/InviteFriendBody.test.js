import React from "react";
import { shallow } from "enzyme";

import InviteFriendBody from "../components/cards/InviteFriendBody";
import Paragraph from "../../../components/Typography/Paragraph";
import { Box } from "@mui/material";
import {
  FONT_WEIGHT_700,
  LINK_N_TEXT_FONT_SIZE,
  PRIMARY_GREEN_COLOR_ACTIVE,
} from "../../../data/constants";

const styledCompleted = {
  fontSize: LINK_N_TEXT_FONT_SIZE,
};

const highlightedText = {
  color: PRIMARY_GREEN_COLOR_ACTIVE,
  fontSize: LINK_N_TEXT_FONT_SIZE,
  fontWeight: FONT_WEIGHT_700,
};

describe("<InviteFriendBody />", () => {
  describe("render()", () => {
    it("Shallow rendering - MatchSnapshot", () => {
      const component = shallow(<InviteFriendBody />);
      expect(component).toMatchSnapshot();
    });

    it("Shallow rendering - contains", () => {
      const component = shallow(<InviteFriendBody />);
      expect(
        component.containsMatchingElement(
          <Paragraph styleProps={{ ...styledCompleted, padding: "1em 0" }}>
            <Box component="span" {...highlightedText} pr={0.5}>
              Receive 50 products
            </Box>
            by inviting a friend who subscribes to a plan. Your friend will
            receive a 30% discount coupon valid for any plan.
          </Paragraph>
        )
      ).toBeTruthy();
    });
  });
});
