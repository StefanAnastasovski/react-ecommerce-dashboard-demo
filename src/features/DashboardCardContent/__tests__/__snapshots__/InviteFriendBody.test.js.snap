// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InviteFriendBody /> render() Shallow rendering - MatchSnapshot 1`] = `ShallowWrapper {}`;
